src/CppParser/Parser.cpp

```diff
@@ -1227,15 +1227,18 @@ Parser::WalkTemplateArgumentList(const clang::TemplateArgumentList* TAL,
 {
     using namespace clang;
 
+	auto LocValid = TSTL && !TSTL->isNull();
+	auto LocTypePtrValid = LocValid && TSTL->getTypePtr() != nullptr;
+
     auto params = std::vector<CppSharp::CppParser::TemplateArgument>();
-    auto typeLocNumArgs = TSTL && TSTL->getTypePtr() ? TSTL->getNumArgs() : 0;
```

Comment:
The entire fix can be much simpler. Just add && !TSTL->isNull() at this line.

Reply:
Yes I know but I wanted to keep the same syntax used on another point of the code where it did a similar thing. If you prefer I can change like you said.

Reply:
Fair enough. Just make sure you use 4 spaces rather than tabs.